src/Datasource/EntityTrait.php

```diff
@@ -281,8 +299,10 @@ public function &get($property)
 
         if ($this->_methodExists($method)) {
             $result = $this->{$method}($value);
-            return $result;
+            return $recursor($result, $nested);
         }
+
+        $value = $recursor($value, $nested);
```

Comment:
Probably yes. My first goal was to make the minimum modifications necessary to maintain complete backwards compatibility and have the test suite pass. The next step is to build the test suite out against this initial implementation to "cement" the expectations. Then as a final step it becomes safe to refactor the original code, using the new tests to ensure any changes still conform to the same expectations. The tests I'm adding are generally going well, though I won't have time to finish them tonight. That means probably tomorrow I expect to be able to submit an updated algorithm that short-circuits as quickly as possible in all cases.